tool/tctl/common/access_request_command.go

```diff
@@ -104,17 +107,25 @@ func (c *AccessRequestCommand) List(client auth.ClientI) error {
 }
 
 func (c *AccessRequestCommand) Approve(client auth.ClientI) error {
+	ctx := context.TODO()
+	if c.delegator != "" {
+		ctx = auth.WithDelegator(ctx, c.delegator)
+	}
```

Comment:
Why pass additional data through the context? Why not just as a parameter to the function?

Reply:
Mostly because `delegator` is intended to function as a caller-definable version of `update_by`, which is also passed by context.
Both `delegator` and `update_by` are consumed solely by the audit log, which means that including them in the definition of the `DynamicAccess` service would be confusing.  Both the client-side and backend implementations of `DynamicAccess` are unable to consume `update_by`.  While `delegator` can be consumed by the client-side implementation, it similarly isn't part of the "standard" DynamicAccess service defintion.  Seemed simpler to keep `delegator` usage consistent with `update_by`, and it seemed consistent with how other authentication data is handled when it needs to be passed across a method that is part of a `services` interface.